### PR TITLE
Remove support for PoisonPill in ActorRefSource (#26622)

### PR DESCRIPTION
**Pull request by nvollmar**
References #26622

### CODE REVIEW

File: akka-stream/src/main/mima-filters/2.6.x.backwards.excludes/26622-stagelogic-internal-api.excludes
Comment (johanandren) [inline akka-stream/src/main/mima-filters/2.6.x.backwards.excludes/26622-stagelogic-internal-api.excludes:2]: Shouldn't this be in the directory 2.6.10.backwards.excludes rather?

**Review by johanandren — COMMENTED** [review]
Looks good, one comment on mima directory.

File: akka-stream/src/main/mima-filters/2.6.x.backwards.excludes/26622-stagelogic-internal-api.excludes
Comment (nvollmar) [reply akka-stream/src/main/mima-filters/2.6.x.backwards.excludes/26622-stagelogic-internal-api.excludes:2]: I wasn't quite sure, I assumed it should be in the directory for release the change will be included?

Comment (johanandren) [reply akka-stream/src/main/mima-filters/2.6.x.backwards.excludes/26622-stagelogic-internal-api.excludes:2]: Ah, we did release 2.6.11 last week, so that's the directory it should be in, I forgot that. 

You can think of it as "binary incompabilities (to ignore) compared to this version", so always the last previous release.

Comment (nvollmar) [reply akka-stream/src/main/mima-filters/2.6.x.backwards.excludes/26622-stagelogic-internal-api.excludes:2]: ok, I can move that accordingly

Comment (nvollmar) [reply akka-stream/src/main/mima-filters/2.6.x.backwards.excludes/26622-stagelogic-internal-api.excludes]: I also moved `29586-sink-internal-api.excludes` to `2.10` as it was released with `2.11`

**Review by johanandren — APPROVED** [review]
LGTM, thanks!

File: akka-stream/src/main/scala/akka/stream/scaladsl/Source.scala
Comment (patriknw) [inline akka-stream/src/main/scala/akka/stream/scaladsl/Source.scala:680]: It's somewhat scary to change behavior in a patch release even though we have had it deprecated for 1 year. What will happen if PoisonPill is still used?

Comment (nvollmar) [reply akka-stream/src/main/scala/akka/stream/scaladsl/Source.scala:680]: Without testing I'd expect the stream actor to terminate, but the stream to stay open.

Comment (johanandren) [reply akka-stream/src/main/scala/akka/stream/scaladsl/Source.scala:680]: Note to ourselves if we decide we shouldn't: we probably shouldn't mark an issue as triaged and assign it to a milestone if we don't think it should be done until the next minor release. ;)

Comment (nvollmar) [reply akka-stream/src/main/scala/akka/stream/scaladsl/Source.scala:680]: This might be better suited for Milestone 2.7 or 3.0

Comment (patriknw) [reply akka-stream/src/main/scala/akka/stream/scaladsl/Source.scala:680]: We can absolutely do it in 2.7.0. That's when we normally remove deprecated methods and change configuration.

I would like that we think some more about it before we close (defer) the PR. I might be overly conservative. It's more than one year of deprecation warning.

Comment (nvollmar) [reply akka-stream/src/main/scala/akka/stream/scaladsl/Source.scala:680]: The deprecation and warning have been added in Akka 2.5.22 almost 2 years ago. One could argue that if someone didn't notice it till now, delaying to Akka 2.7 will probably not change that.

Comment (patriknw) [reply akka-stream/src/main/scala/akka/stream/scaladsl/Source.scala:680]: Thanks for that info. Good that someone is on top of things here ;-)
I'd say that we go ahead and merge this after reviews.

Comment (nvollmar) [reply akka-stream/src/main/scala/akka/stream/scaladsl/Source.scala:680]: Just missing that second review ;-)

**Review by patriknw — APPROVED** [review]
LGTM

**Comment by patriknw** [comment]
OK ΤO ΤESΤ

**Comment by patriknw** [comment]
PLS BUILD

File: akka-stream/src/main/mima-filters/2.6.11.backwards.excludes/26622-stagelogic-internal-api.excludes
Comment (patriknw) [inline akka-stream/src/main/mima-filters/2.6.11.backwards.excludes/26622-stagelogic-internal-api.excludes:1]: this should now be renamed to 2.6.12

Comment (nvollmar) [reply akka-stream/src/main/mima-filters/2.6.11.backwards.excludes/26622-stagelogic-internal-api.excludes]: rebased to latest master and moved file to 2.6.12

**Comment by nvollmar** [comment]
@patriknw can you trigger the build?

**Comment by johanandren** [comment]
PLS BUILD

**Pull request merged**
